lib/serverHandler.dart

```diff
@@ -116,13 +115,15 @@ class ServerHandler {
       data["position"][0].toDouble(),
       data["position"][1].toDouble(),
     );
+    bool isSpectating = data["isSpectating"];
 
     serverData["players"][player]["spaceship"] = data;
 
+    launcher.game.players[player]["spaceship"].isSpectating = isSpectating;
+
     if (player != id) {
       if (launcher.game.players.containsKey(player)) {
-        launcher.game.players[player]["spaceship"].body
-            .setTransform(position, angle);
+        launcher.game.players[player]["spaceship"].body.position = position;
```

Comment:
Are you sure this would update the position for foreign spaceships? I don't see the position variable being used anywhere other than spaceship initialization.

Reply:
You were actually right about this. There's no setter for `body.position`. You were right to use `setTransform()`. It didn't give me a compile error here just because it didn't know what type it was dealing with and a runtime error was never raised because this code never actually got executed given that I never tested with more than one player. 
Now I changed it to `launcher.game.players[player]["spaceship"].body.setTransform(position, 0);`. The angle is set to 0 because we never rotate the body itself, but rather the sprite.